scss/_base_code.scss

```diff
@@ -10,30 +10,32 @@
     text-align: left;
   }
 
+  code {
+    @extend %default-text;
+    margin-bottom: - $baseline-nudge-p;
+  }
+
   pre,
   code {
     direction: ltr;
+    display: block;
     hyphens: none;
-    line-height: 1.5;
-    margin-bottom: 0;
-    margin-top: 0;
     tab-size: 4;
     white-space: pre-wrap;
     word-spacing: normal;
     word-wrap: break-word;
-
-    & + & {
-      @include vf-default-vertical-spacing;
-    }
   }
 
   pre {
     background-color: $color-light;
     border: 1px solid $color-mid-light;
-    border-radius: 2px;
+    border-radius: $border-radius;
     color: $color-dark;
+    margin-bottom: $sp-unit * 4 - $px; // border compensations
+    margin-top: -1px;
```

Comment:
compensates for the borders on the <pre>. Since they are outside the border-box, the negative margin makes the overall height a multiple of the base unit. Slightly modified in the latest commit :     margin-bottom: $spv-nudge-compensation - 2 * $px;